File: typescript/docs/examples/common-features/basics/session-keep-alive/session-keep-alive.ts
import { AgentBay, log, logError, CreateSessionParams } from "wuying-agentbay-sdk";

function sleep(ms: number): Promise<void> {
  return new Promise(resolve => setTimeout(resolve, ms));
}

async function main() {
  const apiKey = process.env.AGENTBAY_API_KEY;
  if (!apiKey) {
    throw new Error("AGENTBAY_API_KEY environment variable is not set");
  }

  const agentBay = new AgentBay({ apiKey });
  let session: any = null;

  try {
    const params: CreateSessionParams = {
      imageId: "linux_latest",
      idleReleaseTimeout: 30,
      labels: { example: "session-keep-alive", sdk: "typescript" },
    };

    const create = await agentBay.create(params);
    if (!create.success || !create.session) {
      throw new Error(`Failed to create session: ${create.errorMessage || "Unknown error"}`);
    }
    session = create.session;
    log(`Session ID: ${session.sessionId}`);

    log("Sleeping for 15 seconds...");
    await sleep(15000);

    log("Calling keepAlive() to refresh idle timer...");
    const keepAliveResult = await session.keepAlive();
    log(`keepAlive success: ${keepAliveResult.success}`);
    log(`requestId: ${keepAliveResult.requestId || ""}`);
    if (!keepAliveResult.success) {
      log(`error: ${keepAliveResult.errorMessage || ""}`);
    }
  } catch (error) {
    logError("Error:", error);
  } finally {
    if (session) {
      await agentBay.delete(session);
    }
  }
}

main().catch(err => {
  logError("Error in main:", err);
  process.exit(1);
});

File: python/docs/examples/_async/common-features/basics/session_operations/session_keep_alive.py
"""Session Keep-Alive Example"""
import asyncio
import os
import sys

sys.path.append(
    os.path.dirname(
        os.path.dirname(
            os.path.dirname(
                os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
            )
        )
    )
)

from agentbay import AsyncAgentBay, CreateSessionParams


async def main():
    print("=== Session Keep-Alive ===\n")

    client = AsyncAgentBay()
    session = None

    try:
        params = CreateSessionParams(
            image_id="linux_latest",
            idle_release_timeout=30,
            labels={"example": "session-keep-alive", "sdk": "python-async"},
        )
        session_result = await client.create(params)
        session = session_result.session
        print(f"Session ID: {session.session_id}")

        print("Sleeping for 15 seconds...")
        await asyncio.sleep(15)

        print("Calling keep_alive() to refresh idle timer...")
        keep_alive_result = await session.keep_alive()
        print(f"keep_alive success: {keep_alive_result.success}")
        print(f"request_id: {keep_alive_result.request_id}")
        if not keep_alive_result.success:
            print(f"error: {keep_alive_result.error_message}")

        print("\n=== Completed ===")
    finally:
        if session:
            await client.delete(session)


if __name__ == "__main__":
    asyncio.run(main())

